conans/test/integration/command_v2/custom_commands_test.py

```diff
@@ -286,3 +288,84 @@ def hello(conan_api, parser, *args, **kwargs):
         # Without the variable it only loads the default custom commands location
         client.run("hello")
         assert "Hello world!" in client.out
+
+    def test_command_reuse_interface(self):
+        mycommand = textwrap.dedent("""
+            import json
+            from conan.cli.command import conan_command
+            from conan.api.output import cli_out_write
+
+            @conan_command(group="custom commands")
+            def mycommand(conan_api, parser, *args, **kwargs):
+                \"""
+                mycommand help
+                \"""
+                result = conan_api.command.run("list", "*", "-c")
+                cli_out_write(json.dumps(result["results"], indent=2))
+            """)
+
+        c = TestClient()
+        command_file_path = os.path.join(c.cache_folder, 'extensions',
+                                         'commands', 'cmd_mycommand.py')
+        c.save({f"{command_file_path}": mycommand})
+        c.run("mycommand", redirect_stdout="file.json")
+        assert json.loads(c.load("file.json")) == {"Local Cache": {}}
+
+    def test_command_reuse_interface_create(self):
+        conanfile = textwrap.dedent("""
+            from conan import ConanFile
+
+            class mylib(ConanFile):
+                name = "mylib"
+            """)
+        mycommand = textwrap.dedent("""
+            import json
+            from conan.cli.command import conan_command
+            from conan.cli.formatters.graph import format_graph_json
+
+            @conan_command(group="custom commands", formatters={"json": format_graph_json})
+            def mycommand(conan_api, parser, *args, **kwargs):
+                \"""
+                mycommand help
+                \"""
+                result = conan_api.command.run("create", ".", "--version=1.0.0")
```

Comment:
this fails if expressed this way, but passes if we call it with the arguments as a list
```
result = conan_api.command.run("create", [".", "--version=1.0.0"])
```

which seems to be a different syntax than the test above that passes them as separate arguments